v2/lib/core/App.js

```diff
@@ -27,23 +27,13 @@ const data = {
   siteConfig,
 };
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      setContext: context => {
```

Comment:
I intentionally call it `setContext` so we can set stuff other than metadata, which we might need in future.